src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -252,9 +252,11 @@ void FFN<
   }
   else if (inputDimensions.size() > 0)
   {
-    const size_t inputDims = std::accumulate(inputDimensions.begin(),
-        inputDimensions.end(), 0);
-    CheckNetwork("FFN::Reset()", inputDims, true, false);
+    size_t inputDim = inputDimensions[0];
+    for (size_t i=1; i<inputDimensions.size(); i++) {
```

Comment:
```suggestion
    for (size_t i = 1; i < inputDimensions.size(); i++)
```
Also, thanks, this is a nice catch!